### ISSUE TITLE
Steam fails to launch: "failed to initialize update status ui, or create initial window"

### ISSUE THREAD

**Issue by siddharthist** [NixOS/nixpkgs#21532]
## Issue description
I installed steam, attempted to launch it, and recieved the following error:
```
Running Steam on nixos 17.03pre97896.d15c62a 64-bit
STEAM_RUNTIME has been set by the user to: /steamrt
Installing breakpad exception handler for appid(steam)/version(0)
No protocol specified
No protocol specified
Assert( Assertion Failed: Could not open connection to X ):updateui_xwin.cpp:337

sh: /home/siddharthist/.local/share/Steam/steam_msg.sh: No such file or directory
Installing breakpad exception handler for appid(steam)/version(1.0)
crash_20161230222639_3.dmp[2343]: Uploading dump (out-of-process)
/tmp/dumps/crash_20161230222639_3.dmp
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2016-12-30 22:26:39] Startup - updater built Nov 17 2015 12:33:46
updateui_xwin.cpp (337) : Assertion Failed: Could not open connection to X
main.cpp (514) : Assertion Failed: failed to initialize update status ui, or create initial window
```

### Steps to reproduce
 1. install steam
 2. run `steam`

## Technical details

* System: 17.03pre97896.d15c62a
* Nix version: 1.11.4
* Nixpkgs version: 17.03pre97896.d15c62a

**Comment by siddharthist** [comment]
This was partially solved by running `xhost +`. I then got the following error:
```
STEAM_RUNTIME has been set by the user to: /steamrt
Installing breakpad exception handler for appid(steam)/version(0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2016-12-30 23:20:48] Startup - updater built Nov 17 2015 12:33:46
[2016-12-30 23:20:48] Verifying installation...
[2016-12-30 23:20:48] Unable to read and verify install manifest /home/siddharthist/.local/share/Steam/package/steam_client_ubuntu12.installed
[2016-12-30 23:20:48] Verification complete
[2016-12-30 23:20:48] Downloading Update...
[2016-12-30 23:20:48] Checking for available update...
[2016-12-30 23:20:48] Package file tenfoot_misc_all.zip.008ed314d467eecacba58d188fdf082d23f264d2 missing or incorrect size
[2016-12-30 23:20:48] Package file tenfoot_dicts_all.zip.87bb49891925214df2d745e12a53518fd2ea86f0 missing or incorrect size
[2016-12-30 23:20:48] Package file tenfoot_fonts_all.zip.vz.7673e4cd32b6752bc621d8bc1a7118a9af19b64a_12077027 missing or incorrect size
[2016-12-30 23:20:48] Package file tenfoot_ambientsounds_all.zip.12963b9dbc3d331e26559cc2d323ca0661942b6b missing or incorrect size
[2016-12-30 23:20:48] Package file tenfoot_sounds_all.zip.vz.ffef2b2fc386819a842ea79484b966a937c2ca7e_1209792 missing or incorrect size
[2016-12-30 23:20:48] Package file tenfoot_images_all.zip.vz.4dd7ec77bcc12bd233175671226254d848147ec5_28235541 missing or incorrect size
[2016-12-30 23:20:48] Package file tenfoot_all.zip.vz.53a00bf41c609500c98cf5d2ef6bd1c1087f4691_2229371 missing or incorrect size
[2016-12-30 23:20:48] Package file resources_misc_all.zip.vz.6596f073f874e66a5af7a41ca94e54e1b091fcb7_2030550 missing or incorrect size
[2016-12-30 23:20:48] Package file resources_all.zip.vz.e4473120a4f4551fe56a1c0b4fb67259933019ea_5053566 missing or incorrect size
[2016-12-30 23:20:48] Package file strings_en_all.zip.vz.430ef651da87a856d87cc1ae9df73b255b53d33d_85768 missing or incorrect size
[2016-12-30 23:20:48] Package file strings_all.zip.vz.7fb608e8bee52eaa93413f066c3aa984bcdc5388_2076553 missing or incorrect size
[2016-12-30 23:20:48] Package file remoteui_all.zip.vz.36b190ff2ea81b6665e7a2bcc98d80c00a9a3517_141317 missing or incorrect size
[2016-12-30 23:20:48] Package file public_all.zip.vz.76d8ac3a090957ead723cef674260b24184fa047_830003 missing or incorrect size
[2016-12-30 23:20:48] Package file bins_ubuntu12.zip.vz.21c1b9865c141a954691e8a4558daa45ee5713ee_16145116 missing or incorrect size
[2016-12-30 23:20:48] Package file bins_pub_ubuntu12.zip.vz.207fdba7daab5691c3355ca74bde4eb63999ab34_9045857 missing or incorrect size
[2016-12-30 23:20:48] Package file bins_sdk_ubuntu12.zip.vz.1fabfcfbaee0698bf1d829ee4f6e9e6cf0c7d76d_9967139 missing or incorrect size
[2016-12-30 23:20:48] Package file bins_codecs_ubuntu12.zip.vz.84b14918e1591ffd02e819b8f7fc79315dbedb03_1832666 missing or incorrect size
[2016-12-30 23:20:48] Package file bins_misc_ubuntu12.zip.4c639ab22a1fe076ff2b705be55ee59f4c6b1122 missing or incorrect size
[2016-12-30 23:20:48] Package file webkit_ubuntu12.zip.vz.4b0455063e3d8b3fe37e2661246835dc4c7d4eaa_42558341 missing or incorrect size
[2016-12-30 23:20:48] Package file miles_ubuntu12.zip.vz.599a10fccbf9ff26130449b95173fbe0cd244549_299828 missing or incorrect size
[2016-12-30 23:20:48] Package file sdl2_ubuntu12.zip.vz.557cbdb00758307b2f34f194fca24bb767ce5bae_270715 missing or incorrect size
[2016-12-30 23:20:48] Package file steam_ubuntu12.zip.vz.f88d31355cb1eb797f4a3bcd268c883701f47f02_1386085 missing or incorrect size
[2016-12-30 23:20:48] Package file runtime_part0_ubuntu12.zip.35d62c2980b8e48d487140919593c82ae5c4a722 missing or incorrect size
[2016-12-30 23:20:48] Package file runtime_part1_ubuntu12.zip.fac99f2e6dbe5e4463336b8269942905848f3e26 missing or incorrect size
[2016-12-30 23:20:48] Downloading update (0 of 266,311 KB)...
[2016-12-30 23:20:48] Downloading update (651 of 266,311 KB)...
[2016-12-30 23:20:49] Downloading update (1,689 of 266,311 KB)...
[2016-12-30 23:20:49] Downloading update (3,375 of 266,311 KB)...
[2016-12-30 23:20:49] Downloading update (5,253 of 266,311 KB)...
[2016-12-30 23:20:49] Downloading update (7,171 of 266,311 KB)...
[2016-12-30 23:20:50] Downloading update (9,036 of 266,311 KB)...
[2016-12-Installing breakpad exception handler for appid(steam)/version(1.0)
30 23:20:50] Downloading update (11,163 of 266,311 KB)...
[2016-12-30 23:20:50] Downloading update (13,511 of 266,311 KB)...
[2016-12-30 23:20:50] Downloading update (15,576 of 266,311 KB)...
[2016-12-30 23:20:51] Downloading update (17,824 of 266,311 KB)...
[2016-12-30 23:20:51] Downloading update (19,864 of 266,311 KB)...
[2016-12-30 23:20:51] Downloading update (21,847 of 266,311 KB)...
[2016-12-30 23:20:51] Downloading update (22,841 of 266,311 KB)...
[2016-12-30 23:20:52] Downloading update (24,379 of 266,311 KB)...
[2016-12-30 23:20:52] Downloading update (25,774 of 266,311 KB)...
[2016-12-30 23:20:52] Downloading update (27,398 of 266,311 KB)...
[2016-12-30 23:20:52] Downloading update (28,661 of 266,311 KB)...
[2016-12-30 23:20:53] Downloading update (30,354 of 266,311 KB)...
[2016-12-30 23:20:53] Downloading update (31,590 of 266,311 KB)...
[2016-12-30 23:20:53] Downloading update (33,158 of 266,311 KB)...
[2016-12-30 23:20:53] Downloading update (34,859 of 266,311 KB)...
[2016-12-30 23:20:54] Downloading update (36,306 of 266,311 KB)...
[2016-12-30 23:20:54] Downloading update (37,765 of 266,311 KB)...
[2016-12-30 23:20:54] Downloading update (39,473 of 266,311 KB)...
[2016-12-30 23:20:54] Downloading update (40,951 of 266,311 KB)...
[2016-12-30 23:20:55] Downloading update (42,271 of 266,311 KB)...
[2016-12-30 23:20:55] Downloading update (43,452 of 266,311 KB)...
[2016-12-30 23:20:55] Downloading update (45,175 of 266,311 KB)...
[2016-12-30 23:20:55] Downloading update (46,323 of 266,311 KB)...
[2016-12-30 23:20:56] Downloading update (47,230 of 266,311 KB)...
[2016-12-30 23:20:56] Downloading update (47,893 of 266,311 KB)...
[2016-12-30 23:20:56] Downloading update (48,410 of 266,311 KB)...
[2016-12-30 23:20:56] Downloading update (49,345 of 266,311 KB)...
[2016-12-30 23:20:57] Downloading update (50,104 of 266,311 KB)...
[2016-12-30 23:20:57] Downloading update (50,745 of 266,311 KB)...
[2016-12-30 23:20:57] Downloading update (51,456 of 266,311 KB)...
[2016-12-30 23:20:57] Downloading update (51,607 of 266,311 KB)...
[2016-12-30 23:20:58] Downloading update (52,516 of 266,311 KB)...
[2016-12-30 23:20:58] Downloading update (53,340 of 266,311 KB)...
[2016-12-30 23:20:58] Downloading update (54,190 of 266,311 KB)...
[2016-12-30 23:20:59] Downloading update (54,675 of 266,311 KB)...
[2016-12-30 23:20:59] Downloading update (55,249 of 266,311 KB)...
[2016-12-30 23:20:59] Downloading update (55,744 of 266,311 KB)...
[2016-12-30 23:20:59] Downloading update (56,456 of 266,311 KB)...
[2016-12-30 23:21:00] Downloading update (57,153 of 266,311 KB)...
[2016-12-30 23:21:00] Downloading update (57,501 of 266,311 KB)...
[2016-12-30 23:21:00] Downloading update (57,927 of 266,311 KB)...
[2016-12-30 23:21:00] Downloading update (58,137 of 266,311 KB)...
[2016-12-30 23:21:01] Downloading update (58,348 of 266,311 KB)...
[2016-12-30 23:21:01] Downloading update (58,569 of 266,311 KB)...
[2016-12-30 23:21:01] Downloading update (58,781 of 266,311 KB)...
[2016-12-30 23:21:02] Downloading update (58,916 of 266,311 KB)...
[2016-12-30 23:21:02] Downloading update (59,419 of 266,311 KB)...
[2016-12-30 23:21:03] Downloading update (60,059 of 266,311 KB)...
[2016-12-30 23:21:03] Downloading update (60,211 of 266,311 KB)...
[2016-12-30 23:21:03] Downloading update (60,437 of 266,311 KB)...
[2016-12-30 23:21:04] Downloading update (60,962 of 266,311 KB)...
[2016-12-30 23:21:04] Downloading update (61,223 of 266,311 KB)...
[2016-12-30 23:21:04] Downloading update (61,783 of 266,311 KB)...
[2016-12-30 23:21:05] Downloading update (62,543 of 266,311 KB)...
[2016-12-30 23:21:05] Downloading update (63,279 of 266,311 KB)...
[2016-12-30 23:21:05] Downloading update (64,015 of 266,311 KB)...
[2016-12-30 23:21:05] Downloading update (65,018 of 266,311 KB)...
[2016-12-30 23:21:06] Downloading update (65,620 of 266,311 KB)...
[2016-12-30 23:21:06] Downloading update (66,187 of 266,311 KB)...
[2016-12-30 23:21:06] Downloading update (66,833 of 266,311 KB)...
[2016-12-30 23:21:06] Downloading update (67,569 of 266,311 KB)...
[2016-12-30 23:21:07] Downloading update (68,168 of 266,311 KB)...
[2016-12-30 23:21:07] Downloading update (69,028 of 266,311 KB)...
[2016-12-30 23:21:07] Downloading update (70,169 of 266,311 KB)...
[2016-12-30 23:21:07] Downloading update (71,159 of 266,311 KB)...
[2016-12-30 23:21:08] Downloading update (72,237 of 266,311 KB)...
[2016-12-30 23:21:08] Downloading update (73,271 of 266,311 KB)...
[2016-12-30 23:21:08] Downloading update (74,396 of 266,311 KB)...
[2016-12-30 23:21:08] Downloading update (75,895 of 266,311 KB)...
[2016-12-30 23:21:09] Downloading update (76,856 of 266,311 KB)...
[2016-12-30 23:21:09] Downloading update (77,899 of 266,311 KB)...
[2016-12-30 23:21:09] Downloading update (79,081 of 266,311 KB)...
[2016-12-30 23:21:09] Downloading update (80,414 of 266,311 KB)...
[2016-12-30 23:21:10] Downloading update (81,298 of 266,311 KB)...
[2016-12-30 23:21:10] Downloading update (82,475 of 266,311 KB)...
[2016-12-30 23:21:10] Downloading update (83,500 of 266,311 KB)...
[2016-12-30 23:21:10] Downloading update (84,770 of 266,311 KB)...
[2016-12-30 23:21:11] Downloading update (85,963 of 266,311 KB)...
[2016-12-30 23:21:11] Downloading update (86,857 of 266,311 KB)...
[2016-12-30 23:21:11] Downloading update (87,833 of 266,311 KB)...
[2016-12-30 23:21:11] Downloading update (88,359 of 266,311 KB)...
[2016-12-30 23:21:12] Downloading update (89,256 of 266,311 KB)...
[2016-12-30 23:21:12] Downloading update (90,536 of 266,311 KB)...
[2016-12-30 23:21:12] Downloading update (91,461 of 266,311 KB)...
[2016-12-30 23:21:12] Downloading update (92,520 of 266,311 KB)...
[2016-12-30 23:21:13] Downloading update (93,276 of 266,311 KB)...
[2016-12-30 23:21:13] Downloading update (94,027 of 266,311 KB)...
[2016-12-30 23:21:13] Downloading update (94,431 of 266,311 KB)...
[2016-12-30 23:21:13] Downloading update (94,881 of 266,311 KB)...
[2016-12-30 23:21:14] Downloading update (95,525 of 266,311 KB)...
[2016-12-30 23:21:14] Downloading update (96,343 of 266,311 KB)...
[2016-12-30 23:21:14] Downloading update (97,157 of 266,311 KB)...
[2016-12-30 23:21:14] Downloading update (97,540 of 266,311 KB)...
[2016-12-30 23:21:15] Downloading update (98,546 of 266,311 KB)...
[2016-12-30 23:21:15] Downloading update (99,496 of 266,311 KB)...
[2016-12-30 23:21:15] Downloading update (100,284 of 266,311 KB)...
[2016-12-30 23:21:15] Downloading update (100,936 of 266,311 KB)...
[2016-12-30 23:21:16] Downloading update (101,521 of 266,311 KB)...
[2016-12-30 23:21:16] Downloading update (101,988 of 266,311 KB)...
[2016-12-30 23:21:16] Downloading update (102,780 of 266,311 KB)...
[2016-12-30 23:21:16] Downloading update (103,632 of 266,311 KB)...
[2016-12-30 23:21:17] Downloading update (104,227 of 266,311 KB)...
[2016-12-30 23:21:17] Downloading update (104,652 of 266,311 KB)...
[2016-12-30 23:21:17] Downloading update (105,704 of 266,311 KB)...
[2016-12-30 23:21:17] Downloading update (106,741 of 266,311 KB)...
[2016-12-30 23:21:18] Downloading update (107,818 of 266,311 KB)...
[2016-12-30 23:21:18] Downloading update (108,826 of 266,311 KB)...
[2016-12-30 23:21:18] Downloading update (109,858 of 266,311 KB)...
[2016-12-30 23:21:18] Downloading update (110,823 of 266,311 KB)...
[2016-12-30 23:21:19] Downloading update (111,818 of 266,311 KB)...
[2016-12-30 23:21:19] Downloading update (112,478 of 266,311 KB)...
[2016-12-30 23:21:19] Downloading update (113,297 of 266,311 KB)...
[2016-12-30 23:21:19] Downloading update (114,282 of 266,311 KB)...
[2016-12-30 23:21:20] Downloading update (115,376 of 266,311 KB)...
[2016-12-30 23:21:20] Downloading update (116,316 of 266,311 KB)...
[2016-12-30 23:21:20] Downloading update (117,278 of 266,311 KB)...
[2016-12-30 23:21:20] Downloading update (118,614 of 266,311 KB)...
[2016-12-30 23:21:21] Downloading update (119,387 of 266,311 KB)...
[2016-12-30 23:21:21] Downloading update (120,071 of 266,311 KB)...
[2016-12-30 23:21:21] Downloading update (120,825 of 266,311 KB)...
[2016-12-30 23:21:21] Downloading update (121,980 of 266,311 KB)...
[2016-12-30 23:21:22] Downloading update (122,759 of 266,311 KB)...
[2016-12-30 23:21:22] Downloading update (123,424 of 266,311 KB)...
[2016-12-30 23:21:22] Downloading update (124,858 of 266,311 KB)...
[2016-12-30 23:21:22] Downloading update (125,636 of 266,311 KB)...
[2016-12-30 23:21:23] Downloading update (126,523 of 266,311 KB)...
[2016-12-30 23:21:23] Downloading update (127,454 of 266,311 KB)...
[2016-12-30 23:21:23] Downloading update (128,193 of 266,311 KB)...
[2016-12-30 23:21:23] Downloading update (128,738 of 266,311 KB)...
[2016-12-30 23:21:24] Downloading update (129,780 of 266,311 KB)...
[2016-12-30 23:21:24] Downloading update (131,229 of 266,311 KB)...
[2016-12-30 23:21:24] Downloading update (132,198 of 266,311 KB)...
[2016-12-30 23:21:24] Downloading update (133,241 of 266,311 KB)...
[2016-12-30 23:21:25] Downloading update (134,309 of 266,311 KB)...
[2016-12-30 23:21:25] Downloading update (135,484 of 266,311 KB)...
[2016-12-30 23:21:25] Downloading update (136,387 of 266,311 KB)...
[2016-12-30 23:21:25] Downloading update (137,328 of 266,311 KB)...
[2016-12-30 23:21:26] Downloading update (138,173 of 266,311 KB)...
[2016-12-30 23:21:26] Downloading update (139,083 of 266,311 KB)...
[2016-12-30 23:21:26] Downloading update (140,153 of 266,311 KB)...
[2016-12-30 23:21:26] Downloading update (141,314 of 266,311 KB)...
[2016-12-30 23:21:27] Downloading update (142,261 of 266,311 KB)...
[2016-12-30 23:21:27] Downloading update (143,005 of 266,311 KB)...
[2016-12-30 23:21:27] Downloading update (143,432 of 266,311 KB)...
[2016-12-30 23:21:27] Downloading update (144,177 of 266,311 KB)...
[2016-12-30 23:21:28] Downloading update (145,254 of 266,311 KB)...
[2016-12-30 23:21:28] Downloading update (146,394 of 266,311 KB)...
[2016-12-30 23:21:28] Downloading update (147,642 of 266,311 KB)...
[2016-12-30 23:21:29] Downloading update (148,605 of 266,311 KB)...
[2016-12-30 23:21:29] Downloading update (149,225 of 266,311 KB)...
[2016-12-30 23:21:29] Downloading update (150,194 of 266,311 KB)...
[2016-12-30 23:21:29] Downloading update (150,778 of 266,311 KB)...
[2016-12-30 23:21:30] Downloading update (151,499 of 266,311 KB)...
[2016-12-30 23:21:30] Downloading update (152,159 of 266,311 KB)...
[2016-12-30 23:21:30] Downloading update (152,835 of 266,311 KB)...
[2016-12-30 23:21:30] Downloading update (153,539 of 266,311 KB)...
[2016-12-30 23:21:31] Downloading update (154,568 of 266,311 KB)...
[2016-12-30 23:21:31] Downloading update (155,241 of 266,311 KB)...
[2016-12-30 23:21:31] Downloading update (155,768 of 266,311 KB)...
[2016-12-30 23:21:31] Downloading update (156,539 of 266,311 KB)...
[2016-12-30 23:21:32] Downloading update (157,083 of 266,311 KB)...
[2016-12-30 23:21:32] Downloading update (157,875 of 266,311 KB)...
[2016-12-30 23:21:32] Downloading update (158,555 of 266,311 KB)...
[2016-12-30 23:21:32] Downloading update (159,743 of 266,311 KB)...
[2016-12-30 23:21:33] Downloading update (161,482 of 266,311 KB)...
[2016-12-30 23:21:33] Downloading update (163,019 of 266,311 KB)...
[2016-12-30 23:21:33] Downloading update (164,261 of 266,311 KB)...
[2016-12-30 23:21:33] Downloading update (165,504 of 266,311 KB)...
[2016-12-30 23:21:34] Downloading update (166,665 of 266,311 KB)...
[2016-12-30 23:21:34] Downloading update (168,460 of 266,311 KB)...
[2016-12-30 23:21:34] Downloading update (169,980 of 266,311 KB)...
[2016-12-30 23:21:34] Downloading update (171,623 of 266,311 KB)...
[2016-12-30 23:21:35] Downloading update (173,244 of 266,311 KB)...
[2016-12-30 23:21:35] Downloading update (174,386 of 266,311 KB)...
[2016-12-30 23:21:35] Downloading update (175,455 of 266,311 KB)...
[2016-12-30 23:21:35] Downloading update (176,661 of 266,311 KB)...
[2016-12-30 23:21:36] Downloading update (177,822 of 266,311 KB)...
[2016-12-30 23:21:36] Downloading update (179,407 of 266,311 KB)...
[2016-12-30 23:21:36] Downloading update (180,868 of 266,311 KB)...
[2016-12-30 23:21:36] Downloading update (182,549 of 266,311 KB)...
[2016-12-30 23:21:37] Downloading update (184,294 of 266,311 KB)...
[2016-12-30 23:21:37] Downloading update (186,046 of 266,311 KB)...
[2016-12-30 23:21:37] Downloading update (187,848 of 266,311 KB)...
[2016-12-30 23:21:37] Downloading update (189,648 of 266,311 KB)...
[2016-12-30 23:21:38] Downloading update (191,042 of 266,311 KB)...
[2016-12-30 23:21:38] Downloading update (192,225 of 266,311 KB)...
[2016-12-30 23:21:38] Downloading update (193,269 of 266,311 KB)...
[2016-12-30 23:21:38] Downloading update (194,355 of 266,311 KB)...
[2016-12-30 23:21:39] Downloading update (195,468 of 266,311 KB)...
[2016-12-30 23:21:39] Downloading update (196,711 of 266,311 KB)...
[2016-12-30 23:21:39] Downloading update (197,937 of 266,311 KB)...
[2016-12-30 23:21:39] Downloading update (199,241 of 266,311 KB)...
[2016-12-30 23:21:40] Downloading update (200,571 of 266,311 KB)...
[2016-12-30 23:21:40] Downloading update (201,710 of 266,311 KB)...
[2016-12-30 23:21:40] Downloading update (202,875 of 266,311 KB)...
[2016-12-30 23:21:40] Downloading update (203,701 of 266,311 KB)...
[2016-12-30 23:21:41] Downloading update (204,779 of 266,311 KB)...
[2016-12-30 23:21:41] Downloading update (205,641 of 266,311 KB)...
[2016-12-30 23:21:41] Downloading update (206,875 of 266,311 KB)...
[2016-12-30 23:21:41] Downloading update (208,133 of 266,311 KB)...
[2016-12-30 23:21:42] Downloading update (209,810 of 266,311 KB)...
[2016-12-30 23:21:42] Downloading update (211,104 of 266,311 KB)...
[2016-12-30 23:21:42] Downloading update (212,431 of 266,311 KB)...
[2016-12-30 23:21:42] Downloading update (213,868 of 266,311 KB)...
[2016-12-30 23:21:43] Downloading update (215,276 of 266,311 KB)...
[2016-12-30 23:21:43] Downloading update (216,946 of 266,311 KB)...
[2016-12-30 23:21:43] Downloading update (218,312 of 266,311 KB)...
[2016-12-30 23:21:43] Downloading update (220,096 of 266,311 KB)...
[2016-12-30 23:21:44] Downloading update (221,751 of 266,311 KB)...
[2016-12-30 23:21:44] Downloading update (223,549 of 266,311 KB)...
[2016-12-30 23:21:44] Downloading update (225,390 of 266,311 KB)...
[2016-12-30 23:21:44] Downloading update (226,482 of 266,311 KB)...
[2016-12-30 23:21:45] Downloading update (227,810 of 266,311 KB)...
[2016-12-30 23:21:45] Downloading update (229,187 of 266,311 KB)...
[2016-12-30 23:21:45] Downloading update (230,387 of 266,311 KB)...
[2016-12-30 23:21:45] Downloading update (231,598 of 266,311 KB)...
[2016-12-30 23:21:46] Downloading update (232,803 of 266,311 KB)...
[2016-12-30 23:21:46] Downloading update (234,228 of 266,311 KB)...
[2016-12-30 23:21:46] Downloading update (235,909 of 266,311 KB)...
[2016-12-30 23:21:47] Downloading update (237,362 of 266,311 KB)...
[2016-12-30 23:21:47] Downloading update (238,648 of 266,311 KB)...
[2016-12-30 23:21:47] Downloading update (239,742 of 266,311 KB)...
[2016-12-30 23:21:47] Downloading update (240,751 of 266,311 KB)...
[2016-12-30 23:21:48] Downloading update (241,984 of 266,311 KB)...
[2016-12-30 23:21:48] Downloading update (243,041 of 266,311 KB)...
[2016-12-30 23:21:48] Downloading update (244,235 of 266,311 KB)...
[2016-12-30 23:21:48] Downloading update (245,422 of 266,311 KB)...
[2016-12-30 23:21:49] Downloading update (246,620 of 266,311 KB)...
[2016-12-30 23:21:49] Downloading update (247,926 of 266,311 KB)...
[2016-12-30 23:21:49] Downloading update (249,233 of 266,311 KB)...
[2016-12-30 23:21:49] Downloading update (250,494 of 266,311 KB)...
[2016-12-30 23:21:50] Downloading update (251,828 of 266,311 KB)...
[2016-12-30 23:21:50] Downloading update (253,121 of 266,311 KB)...
[2016-12-30 23:21:50] Downloading update (254,404 of 266,311 KB)...
[2016-12-30 23:21:50] Downloading update (255,555 of 266,311 KB)...
[2016-12-30 23:21:51] Downloading update (256,774 of 266,311 KB)...
[2016-12-30 23:21:51] Downloading update (257,948 of 266,311 KB)...
[2016-12-30 23:21:51] Downloading update (258,642 of 266,311 KB)...
[2016-12-30 23:21:51] Downloading update (259,390 of 266,311 KB)...
[2016-12-30 23:21:52] Downloading update (260,054 of 266,311 KB)...
[2016-12-30 23:21:52] Downloading update (260,734 of 266,311 KB)...
[2016-12-30 23:21:52] Downloading update (261,363 of 266,311 KB)...
[2016-12-30 23:21:52] DownloadinInstalling breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
Installing breakpad exception handler for appid(steam)/version(1.0)
g update (262,021 of 266,311 KB)...
[2016-12-30 23:21:53] Downloading update (262,740 of 266,311 KB)...
[2016-12-30 23:21:53] Downloading update (263,454 of 266,311 KB)...
[2016-12-30 23:21:53] Downloading update (264,130 of 266,311 KB)...
[2016-12-30 23:21:53] Downloading update (264,799 of 266,311 KB)...
[2016-12-30 23:21:54] Downloading update (265,447 of 266,311 KB)...
[2016-12-30 23:21:54] Downloading update (266,158 of 266,311 KB)...
[2016-12-30 23:21:54] Downloading update (266,311 of 266,311 KB)...
[2016-12-30 23:21:54] Download Complete.
[2016-12-30 23:21:54] uninstalled manifest found in /home/siddharthist/.local/share/Steam/package/steam_client_ubuntu12 (1).
[2016-12-30 23:21:54] Extracting package...
[2016-12-30 23:22:05] Installing update...
[2016-12-30 23:22:06] Cleaning up...
[2016-12-30 23:22:06] Update complete, launching...
[2016-12-30 23:22:06] Shutdown
Restarting Steam by request...
Running Steam on nixos 17.03pre97896.d15c62a 64-bit
STEAM_RUNTIME has been set by the user to: /steamrt
Error: You are missing the following 32-bit libraries, and Steam may not run:
libGL.so.1
Error:
You are missing the following 32-bit libraries, and Steam may not run:
libGL.so.1
```
which I think I saw on another issue. I'll keep digging.

**Comment by siddharthist** [comment]
Indeed, that solution is documented on https://github.com/NixOS/nixpkgs/issues/19518. Closing, please reopen if you think it's worth documenting the 
```
xhost + 
```
solution.

**Comment by jameysharp** [comment]
Better than `xhost +` would be if the `XAUTHORITY` environment variable were preserved inside the FHS environment. I've verified that if before running a game I manually set that variable to the value it has in my regular shell, then the game works even without `xhost +`. One way to confirm that for yourself is to prefix the invocation of the game you're trying to play like so:

```
steam-run env XAUTHORITY="$XAUTHORITY"
```

Since the effect of `xhost +` is to completely disable security for connections to the X server, that should not be a recommended approach.

Is this just a matter of adding `XAUTHORITY` to `envvars` in `chroot-user.rb`, since `DISPLAY` is already there?
